keps/sig-testing/0014-20180523-test-cluster-framework.md

```diff
@@ -0,0 +1,169 @@
+---
+kep-number: 14
+title: Test Cluster Framework
+authors:
+  - "@timothysc"   # Timothy St. Clair
+  - "@marun"       # Maru Newby
+  - "@liztio"      # Liz Frost
+  - "@frankgreco"  # Frank B Greco Jr
+  - "@hoegaarden"  # Hannes Hoerl
+  - "@totherme"    # Gareth Smith
+  - "@mariantalla" # Maria Ntalla
+owning-sig: sig-testing
+participating-sigs:
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-05-23
+last-updated: 2018-06-19
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Test Cluster Framework
+
+## Table of Contents
+
+* [Test Cluster Framework](#test-cluster-framework)
+   * [Table of Contents](#table-of-contents)
+   * [Summary](#summary)
+   * [Motivation](#motivation)
+      * [Goals](#goals)
+      * [Non-Goals](#non-goals)
+   * [Proposal](#proposal)
+      * [User Stories](#user-stories)
+      * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Risks and Mitigations](#risks-and-mitigations)
+   * [Graduation Criteria](#graduation-criteria)
+   * [Implementation History](#implementation-history)
+
+<!--
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+-->
+
+## Summary
+
+> Making testing with clusters consistent
+
+We propose an abstraction so that users and tools (e.g. CI) have a consistent
+way to spin up and configure clusters they can test against. This abstraction
+is targeted mainly to the needs of the following personas and should work
+equally well used interactively and non-interactively / automated.
+
+Alice, an experienced CLI Dev (see [Key Persona Alice][alice]), finds that she can’t
+easily onboard new contributors like Bob (see [Key Persona Bob][bob]), in part
+because her CLI and tests are too tightly coupled with heavyweight
+infrastructure in `k8s.io/kubernetes`. To solve this problem, Alice sponsors
+the development of a new lightweight testing infrastructure outside
+`k8s.io/kubernetes`.
+
+Meanwhile, Carol, an API/Controller dev (see [Key Persona Carol][carol]), is
+frustrated that there are already too many ways to spin up test clusters.
+
+There is a clear tension between Alice and Carol’s frustrations. Left to their
+own devices, Alice is likely to contribute to the proliferation of test
+infrastructure that frustrates Carol, and Carol may find herself advocating
+that Alice be prevented from acting to resolve her own frustration with the
```

Comment:
Perhaps I've misread this persona section, but it reads to me as somewhat hostile to Alice, which I don't approve of.  

I don't think anything we check in as a KEP should suggest preventing others from developing anything, let alone test infrastructure...? Why can't "Carol" let "Alice" build whatever tooling they want?

Reply:
This whole section beyond "We propose an abstraction so ..." (anything after the first paragraph) seems fairly unnecessary. I'd suggest a motivation more along the lines of "we'd like a standard framework for spinning up clusters for testing with varying requirements and backing implementations" or something along those lines.